src/osd/PrimaryLogPG.cc

```diff
@@ -3717,6 +3717,27 @@ int PrimaryLogPG::do_xattr_cmp_str(int op, string& v1s, bufferlist& xattr)
   }
 }
 
+int PrimaryLogPG::do_checksum(OpContext *ctx, OSDOp& osd_op)
+{
+  auto& op = osd_op.op;
+  std::vector<OSDOp> read_ops(1);
+  auto& read_op = read_ops[0];
+
+  read_op.op.op = CEPH_OSD_OP_SYNC_READ;
+  read_op.op.flags = op.flags;
+  read_op.op.extent = op.extent;
+
+  int r = do_osd_ops(ctx, read_ops);
+  if (r < 0) {
+    derr << __func__ << " do_osd_ops failed: " << cpp_strerror(r) << dendl;
+    return r;
+  }
+
+  uint32_t crc = read_op.outdata.crc32c(-1);
+  ::encode(crc, osd_op.outdata);
```

Comment:
We should try to use Checksummer.h to generalize this to all supported checksum algorithms